Adding note in docs about cmocean cmap changes. (#3207)
yt-project/yt: doc/source/yt4differences.rst

first time in a session, a warning will be logged, so it is advisable to set
your logging level to ``WARNING`` at a minimum to catch these.

Colormaps
^^^^^^^^^

> yt previously included ``cmocean`` colormaps.  Due to collisions between some
> of our dependencies, you now have to explicitly import the ``cmocean`` module
> and prefix ``cmocean`` colormaps (like ``balance``) with ``cmo.``.  Example:

Reviewer: ```suggestion
yt used to automatically register colormaps from ``cmocean``, unprefixed.
This unfortunately became unsustainable with the 3.4 release of Matplotlib, in which colormaps with colliding
names raise errors. The fix is to explicitly import the ``cmocean`` module and prefix ``cmocean`` colormaps
(like ``balance``) with ``cmo.``.  Note that this solution works with any versions of yt+Matplotlib. Example:
```

Reviewer: I originally wrote that your example worked with any version of yt+matplotlib. After working on #3212, I realised this was unfortunately not the case, so I updated this accordingly.

Old Script (no longer works):
.. code-block:: python
import yt
ds = yt.testing.fake_amr_ds()

> p = yt.SlicePlot(ds, "z", "Density")
> p.set_cmap("Density", "balance")
> 
> New Script:
> .. code-block:: python
> import yt
> import cmocean
> ds = yt.testing.fake_amr_ds()
> p = yt.SlicePlot(ds, "z", "Density")
> p.set_cmap("Density", "cmo.balance")

Reviewer: While we're at it, let's avoid adding more ambiguous fields :)
```suggestion
p = yt.SlicePlot(ds, "z", ("gas", "Density"))
p.set_cmap(("gas", "Density"), "balance")

New Script:
.. code-block:: python
import yt
import cmocean
ds = yt.testing.fake_amr_ds()
p = yt.SlicePlot(ds, "z", ("gas", "Density"))
p.set_cmap(("gas", "Density"), "cmo.balance")
```

> Old Script (no longer works):
> .. code-block:: python
> import yt
> ds = yt.testing.fake_amr_ds()
> p = yt.SlicePlot(ds, "z", "Density")
> p.set_cmap("Density", "balance")
> 
> New Script:
> .. code-block:: python
> import yt
> import cmocean
> ds = yt.testing.fake_amr_ds()
> p = yt.SlicePlot(ds, "z", "Density")
> p.set_cmap("Density", "cmo.balance")

Reviewer: I think Sphinx requires some more whitespacing to render code blocks, and while we're at it, let's avoid adding more ambiguous fields :)
```suggestion
Old Script (no longer works):

.. code-block:: python

    import yt

    ds = yt.testing.fake_amr_ds()
    p = yt.SlicePlot(ds, "z", ("gas", "Density"))
    p.set_cmap(("gas", ("gas", "Density")), "balance")

New Script:

.. code-block:: python

    import yt
    import cmocean

    ds = yt.testing.fake_amr_ds()
    p = yt.SlicePlot(ds, "z", ("gas", "Density"))
    p.set_cmap(("gas", "Density"), "cmo.balance")
```

first time in a session, a warning will be logged, so it is advisable to set
your logging level to ``WARNING`` at a minimum to catch these.

Colormaps
^^^^^^^^^
yt used to automatically register colormaps from ``cmocean``, unprefixed.
This unfortunately became unsustainable with the 3.4 release of Matplotlib,
in which colormaps with colliding names raise errors. The fix is to explicitly
import the ``cmocean`` module and prefix ``cmocean`` colormaps (like
``balance``) with ``cmo.``.  Note that this solution works with any
yt-supported version of Matplotlib, but is not backward compatible with earlier
versions of yt. Example:

> Old Script (no longer works):

Reviewer: Following #3214
```suggestion
Old Script (will not work in the near future):
```
